src/audio/asrc/asrc_farrow.c

```diff
@@ -402,23 +403,25 @@ enum asrc_error_code asrc_initialise(struct comp_dev *dev,
 	}
 
 	/*
-	 * Set the pointer for the impulse response.
-	 * &m_impulse_response[0] = ring_buffers_x
-	 * + num_channels * (1 + buffer_length);
-	 * Here ring_buffer_x already points to the memory just behind
-	 * the src_obj.  The offset results from the number of pointers
-	 * pointing to each channel and each channels buffer data.
+	 * Set the pointer for the impulse response. It is just after
+	 * the ring_buffers32[] or ring_buffers16[] array.
 	 */
+
+	/* Error if odd buffer length */
+	if (src_obj->buffer_length & 1) {
```

Comment:
It is always even atm since computed as `src_obj->filter_length * 2` by `initialise_buffer()`. Should this check be moved to `initialise_buffer()` instead, as an assert() with a comment that change to the formula would result in unaligned pointer later?

Reply:
I have in plans to remove the duplicated data delay line so I added this check to not forget it. But you are right it's unnecessary now. I can remove it since you mentioned.